WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsUsageTracksEventEmitter.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/// Note: If we ever change the algorithm in the future, we should probably consider renaming
+/// the Tracks event to avoid incorrect comparisons with old events.
+final class StoreStatsUsageTracksEventEmitter {
+
+    /// TODO Replace with proper injection
+    static let shared = StoreStatsUsageTracksEventEmitter()
+
+    private let analytics: Analytics
+
+    /// The minimum amount of time (seconds) that the merchant have interacted with the
+    /// Analytics UI before an event is triggered.
+    private let minimumInteractionTime: TimeInterval = 10
+
+    /// The minimum number of Analytics UI interactions before an event is triggered.
+    ///
+    /// The interactions captured are:
+    ///
+    /// - Scrolling
+    /// - Pull-to-refresh
+    /// - Tapping on the bars in the chart
+    /// - Changing the tab
+    /// - Navigating to the My Store tab
+    private let interactionsThreshold = 5
+
+    /// The maximum number of seconds in between interactions before we will consider the
+    /// merchant to have been idle. If they were idle, the time and interactions counting
+    /// will be reset.
+    private let idleTimeThreshold: TimeInterval = 10
```

Comment:
I think it would be safe to make this bit longer than the minimum interaction time. Maybe 15 or 20s

Reply:
Increased to 20s in c83325ef6eb0cb12b35308cbb40448327fe17ac2 (draft PR https://github.com/woocommerce/woocommerce-ios/pull/5820)